library/classes/Prescription.class.php

```diff
@@ -265,6 +268,7 @@ function set_unit($unit)
     function set_id($id)
     {
         if (!empty($id) && is_numeric($id)) { $this->id = $id; }
+        $this->id = $id;
```

Comment:
This is original code. I have not changed this. Yes, it is odd but I do not know the reason for it.